src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -497,6 +497,14 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
         newDevice.constraints = proposedConstraints;
       } else {
         newDevice = await this.getDeviceFromBrowser(proposedConstraints);
+        newDevice.stream.getTracks()[0].addEventListener('ended', () => {
```

Comment:
Does device change event fire in this case ?

Reply:
The device change event fires after the `ended` track listener fires. The consuming application can then optionally use that event (via the DeviceController) to prompt the user to select a new device.